### PR TITLE
Move matrix free classes from CUDAWrappers namespace to Portable namespace

### PR DESCRIPTION
**Pull request by Rombur**
This is a proposal to move the matrix free classes out of the CUDAWrappers namespace to a new Portable namespace. The current namespace doesn't make sense anymore because the current code works fine on AMD GPUs. I am open to a different name for the new namespace if someone as a better idea. There is the issue on how to deprecated the old namespace. nvcc seems to have an issue with the deprecated attribute for a namespace.

### CODE REVIEW

File: include/deal.II/matrix_free/cuda_fe_evaluation.h [hunk 1]
```diff
@@ -40,561 +25,7 @@ DEAL_II_NAMESPACE_OPEN
...
-  template <int dim,
-            int fe_degree,
-            int n_q_points_1d,
-            int n_components_,
-            typename Number>
-  constexpr unsigned int
-    FEEvaluation<dim, fe_degree, n_q_points_1d, n_components_, Number>::
-      n_q_points;
-#endif
+  using namespace Portable;
```
Comment (marcfehling) [inline include/deal.II/matrix_free/cuda_fe_evaluation.h:28]: That's clever. Is there also a way to effectively deprecate the whole `CUDAWrappers` namespace?

**Review by marcfehling — COMMENTED** [review]
Looks good to me. I also don't have a better name for the namespace, but I do like that your choice is 4 characters shorter, which gets rid of some linebreaks in clang-format :)

Would you please also add a changelog entry to `incompatibilities`?

File: include/deal.II/matrix_free/cuda_fe_evaluation.h (hunk 1 above)
Comment (Rombur) [reply include/deal.II/matrix_free/cuda_fe_evaluation.h:28]: The problem is that `nvcc` doesn't like when I deprecate the namespace. With the others compilers I can do
`namespace DEAL_II_EARLY_DEPRECATION CUDAWrappers` and it works like expected. We could only emit the deprecation warning when people use a compiler other than `nvcc` but I would expect that almost everybody that uses this class also uses `nvcc`.

Comment (masterleinad) [reply include/deal.II/matrix_free/cuda_fe_evaluation.h:28]: What does "`nvcc` doesn't like when I deprecate the namespace` exactly mean?
Is the attribute just not recognized and the compiler warns about that (like in https://godbolt.org/z/Mf3444sP9 which only recognizes the attribute from `nvcc 11.1.0` on) or is there a different issue?

Comment (Rombur) [reply include/deal.II/matrix_free/cuda_fe_evaluation.h:28]: Yes, the compiler complains that attribute is not recognized.

Comment (masterleinad) [reply include/deal.II/matrix_free/cuda_fe_evaluation.h:28]: We should just do a version check then since it seems to work from `nvcc 11.1.0` on.

Comment (Rombur) [reply include/deal.II/matrix_free/cuda_fe_evaluation.h:35]: I don't think that the issue is due to `nvcc` version.  The CI fails with `nvcc` 11.8. I think that the problem is the underlying gcc compiler.  The CI uses gcc 9 which had a bug when using `[[deprecated]]` with a namespace (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79817)

**Review by kronbichler — APPROVED** [review]

**Comment by Rombur** [comment]
Unless someone has an objection, I will merge this tomorrow.

**Pull request merged**